src/html.js

```diff
@@ -165,23 +151,35 @@ const htmlAsyncGenerator = async function* ({ raw: literals }, ...expressions) {
         }
 
         for await (const value of expressions[index]) {
-          if (typeof value === "string") {
-            expression = value;
-          } else if (value === undefined || value === null) {
+          if (value === undefined || value === null) {
             continue;
-          } else if (
+          }
```

Comment:
Can an expression be a `Promise`?

Reply:
Passing `` htmlAsyncGenerator`${readFile('./staticPartial.html')}` `` is permitted, so yeah